updated dev doc (#25)
Bit-Quill/tinkerpop: docs/src/dev/developer/development-environment.asciidoc

`gremlin-dotnet` module. If .NET SDK is not installed, TinkerPop will still build with Maven, but .NET projects
will be skipped.

> `gremlin-dotnet` uses Docker for all of its testing. Please make sure Docker is installed and running on your system. 

Reviewer: It is better to clarify - for running integration tests against live gremlin server.
```suggestion
`gremlin-dotnet` uses Docker for running integration tests against the Gremlin server. Please make sure Docker is installed and running on your system. 
```

built. Consequently, the scripts `bin/generate-home.sh` and `bin/publish-home.sh` require that Node.js and npm are
installed. Version 6.x or newer of npm is required. This is covered in more detail in the <<site,Site>> section.

> `gremlin-javascript` uses Docker for all of its testing. Please make sure Docker is installed and running on your system. 

Reviewer: Same suggestions as above for dotnet

> `gremlin-javascript` uses Docker for all of its testing. Please make sure Docker is installed and running on your system. 

Author: ```suggestion
`gremlin-javascript` uses Docker for running integration tests against the Gremlin server. Please make sure Docker is installed and running on your system. 
```

To be a bit more clear, the version can not be a Docker tag like "latest" because there is no such TinkerPop artifact
that has been published with that version number.

[[docker-testing]]
== Testing with Docker

> Some GLV allow you to run tests without Gremlin Server build. Currently supported gremlin-go, gremlin-javascript and

Reviewer 2: I'm not sure what this first sentence is trying to explain.

Author: changed in https://github.com/Bit-Quill/tinkerpop/pull/25/commits/b58ec43eb69b2bcc41330a60c92f62cd74b3f5ed

gremlin-dotnet.

Please make sure Docker is installed and running on your system. 
You will need to install both link:https://docs.docker.com/engine/install/[Docker Engine] and link:https://docs.docker.com/compose/install/[Docker Compose], which are included in link:https://docs.docker.com/desktop/[Docker Desktop].

> The docker compose environment variable `GREMLIN_SERVER` specifies the Gremlin server docker image to use, i.e. an

Reviewer: Can we have it as optional? If it is not specified default image will be pulled?

Author: default version implemented only in `gremlin-go/run.sh` as I understand

> The docker compose environment variable `GREMLIN_SERVER` specifies the Gremlin server docker image to use, i.e. an

Reviewer 2: I think it would be good to mention here something like the following:
"The following environment variables used by Docker Compose will automatically be set when running with Maven."

Author: added in https://github.com/Bit-Quill/tinkerpop/pull/25/commits/b58ec43eb69b2bcc41330a60c92f62cd74b3f5ed

image with the tag `tinkerpop/gremlin-server:$GREMLIN_SERVER`, and is a required environment variable. This also 
requires the specified docker image to exist, either locally or in link:https://hub.docker.com/r/tinkerpop/gremlin-server[Docker Hub].

> Running `mvn clean install -pl gremlin-server -DskipTests -DskipIntegrationTests=true -Dci -am` in the main `tinkerpop` 

Reviewer 2: Could probably use a simpler example here as the image should build as long as you don't have -DskipImageBuild=true

Author: I think it's faster to build a server without running all tests

directory will automatically build a local SNAPSHOT Gremlin server image. If your OS Platform cannot build a local 
SNAPSHOT Gremlin server through `maven`, it is recommended to use the latest released server version from link:https://hub.docker.com/r/tinkerpop/gremlin-server[Docker Hub] (do not use `GREMLIN_SERVER=latest`,
use actual version number, e.g. `GREMLIN_SERVER=3.5.x` or `GREMLIN_SERVER=3.6.x`).

The docker compose environment variable `HOME` specifies the user home directory for mounting volumes during test image 

> set up. This varibale is set by default in Unix/Linux, but will need to be set for Windows, for example, run 

Reviewer: ```suggestion
set up. This varibale is set by default in Unix/Linux and macOS, but will need to be set for Windows, for example, run 
```

> set up. This varibale is set by default in Unix/Linux and macOS, but will need to be set for Windows, for example, run 

Reviewer 2: spelling: variable

Author: fixed

To be a bit more clear, the version can not be a Docker tag like "latest" because there is no such TinkerPop artifact
that has been published with that version number.

[[docker-testing]]
== Testing with Docker

> Some GLV allow you to run tests without setting up Java and without building of other parts of the server. Currently

Reviewer 3: Is it true that one doesn't need to set up Java? Because in the instructions below we are telling people to build the gremlin server, which needs Java?

Author: Can you run gremlin-go test without building server?

Reviewer 3: But running anything in Maven needs a Java JDK though?

Reviewer 3: I'd say it's more true that it doesn't need to build server, but Java is still required. And it is specified as an requirement for TinkerPop anyways.

Author: yes, but `run.sh` don't use Maven

Author: or manual `docker-compose up`

Author: fixed

> supported gremlin-go, gremlin-javascript and gremlin-dotnet.

Reviewer 3: We can add the python and console stuff since we'll be PRing this with it

Author: added

> Some GLV allow you to run tests without building of other parts of the server. Currently supported gremlin-go, 
> gremlin-javascript, gremlin-dotnet, gremlin-python and gremlin-console.

Reviewer 3: ```suggestion
Currently gremlin-go, gremlin-javascript, gremlin-dotnet, gremlin-python and gremlin-console can be tested through Docker.
```
